### ISSUE TITLE
Is there any possibility of setting a custom tuple name without having to hack the script with ``sed``?

### ISSUE THREAD

**Issue by takusuman** [firasuke/mussel#39]
G'day!

I was studying CMLFS lately and perceived that it uses [a patch for changing GCC's vendor instead of a variable or even a command line option](https://github.com/dslm4515/CMLFS/blob/master/1-Stage0/1-Mussel#L25-L27). Would it be possible to implement it in a more friendly way or it would garble mussel's argument parser? An environment variable would be fine too.

Much obliged for the attention.

**Comment by takusuman** [comment]
@firasuke Do you prefer it as a environment variable or as a command line option?
I can implement both, but I can't think which is the best.

**Comment by firasuke** [comment]
Hey there!

It shouldn't be an issue, you can use a triplet or a quadruplet, but do keep in mind that `pc` is not the correct vendor for most of the supported architectures.

Tuples are actually quadruplets, and while the vendor is generally `pc`, but for a default value using `unknown` (more common) or `generic` (less common) might be better.

Also `config.guess` and `config.sub` prefer `pc` to `unknown` and not specifying anything, because it's more commonly used.

IIRC `musl-cross-make` does not define a vendor field, while [skarnet](https://skarnet.org/toolchains/) and [musl.cc](https://musl.cc/)) use `linux` as vendor.

In the end, and as `mussel` is a cross-compilation script, we could use `mussel` or a unique value that signals to the build system that we're cross compiling.

**Comment by firasuke** [comment]
> @firasuke Do you prefer it as a environment variable or as a command line option? I can implement both, but I can't think which is the best.

I believe having it as an environment variable would be the better option; we're already relying on the user providing the architecture as an argument to mussel.

**Comment by takusuman** [comment]
> It shouldn't be an issue, you can use a triplet or a quadruplet, but do keep in mind that `pc` is not the correct vendor for most of the supported architectures.

Of course, good to know that. I'm just studying CMLFS for now because I'm having trouble with the LLVM for the first stage, then I noticed that I'm not using a tuple, while CMLFS does it --- not sure about Chimera and other systems that are built with LLVM.

> IIRC musl-cross-make does not define a vendor field, while [skarnet](https://skarnet.org/toolchains/) and [musl.cc](https://musl.cc/) use linux as vendor.

Sure, but I'm sort of confused about why we define it as ``pc`` for building with LLVM... Maybe it's not the reason after all.

> In the end, and as mussel is a cross-compilation script, we could use mussel or a unique value that signals to the build system that we're cross compiling.

Well, keeping it "vanilla", as musl-cross-make, is great, but we could add this option for cases that need to inform that it is a cross-toolchain.

**Comment by takusuman** [comment]
> I believe having it as an environment variable would be the better option; we're already relying on the user providing the architecture as an argument to mussel.

A command line option would be completely optional and would depend on the user providing the architecture before, so there won't be bikeshedding at the argument parser.
